server/src/main/java/org/spine3/server/stand/QueryProcessor.java

```diff
@@ -0,0 +1,223 @@
+import com.google.common.collect.FluentIterable;
+import com.google.common.collect.ImmutableCollection;
+import com.google.common.collect.ImmutableList;
+import com.google.protobuf.Any;
+import com.google.protobuf.FieldMask;
+import com.google.protobuf.Message;
+import io.grpc.stub.StreamObserver;
+import org.spine3.base.Queries;
+import org.spine3.client.EntityFilters;
+import org.spine3.client.EntityId;
+import org.spine3.client.EntityIdFilter;
+import org.spine3.client.Query;
+import org.spine3.client.Target;
+import org.spine3.protobuf.AnyPacker;
+import org.spine3.protobuf.TypeUrl;
+import org.spine3.server.entity.Entity;
+import org.spine3.server.entity.EntityRepository;
+import org.spine3.server.storage.EntityStorageRecord;
+import org.spine3.server.storage.StandStorage;
+
+import javax.annotation.Nullable;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Set;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+/**
+ * Query processing helper utility.
+ *
+ * @author Alex Tymchenko
+ */
+/* package */
+@SuppressWarnings("UtilityClass")
+class QueryProcessor {
+
+    private QueryProcessor() {
+    }
+
+    /**
+     * Performs query processing as a part of {@link Stand#execute(Query, StreamObserver)}.
+     *
+     * @param query               an instance of {@code Query} to process
+     * @param standStorage        internal storage of {@link Stand}
+     * @param aggregateTypes      {@link org.spine3.server.aggregate.Aggregate} types exposed by the {@code Stand}
+     * @param typeToRepositoryMap map of exposed types and related repositories registered in the {@code Stand}
+     * @return the query result
+     */
+    /* package */
+    static ImmutableCollection<Any> processQuery(Query query,
+                                                 StandStorage standStorage,
+                                                 Set<TypeUrl> aggregateTypes,
+                                                 Map<TypeUrl, EntityRepository<?, ? extends Entity, ? extends Message>> typeToRepositoryMap) {
+        final ImmutableList.Builder<Any> resultBuilder = ImmutableList.builder();
+
+        final TypeUrl typeUrl = Queries.typeOf(query);
+        checkNotNull(typeUrl, "Target type unknown");
+
+        final EntityRepository<?, ? extends Entity, ? extends Message> repository = typeToRepositoryMap.get(typeUrl);
+        if (repository != null) {
+
+            // the target references an entity state
+            final ImmutableCollection<? extends Entity> entities = fetchFromEntityRepository(query, repository);
+            feedEntitiesToBuilder(resultBuilder, entities);
+
+        } else if (aggregateTypes.contains(typeUrl)) {
+
+            // the target relates to an {@code Aggregate} state
+            final ImmutableCollection<EntityStorageRecord> stateRecords = fetchFromStandStorage(query, standStorage, typeUrl);
+            feedStateRecordsToBuilder(resultBuilder, stateRecords);
+        }
```

Comment:
Having these classes would isolate modifications when we modify either aggregate or entity storage. Also if we add another kind of entities, it would result in introducing another `Feeder` that would handle the specifics.